doc/sphinx/language/gallina-extensions.rst

```diff
@@ -869,255 +870,297 @@ Module system
+is also used to terminate :ref:`Sections<section-mechanism>`.
+:n:`End @ident` closes the interactive module or module type :token:`ident`.
+If the module type was given, the command verifies that the content of the module
+matches the module type.  If the module is not a
+functor, its components (constants, inductive types, submodules etc.)
+are now available through the dot notation.
 
-.. cmd:: Include @module
+.. exn:: No such label @ident.
+    :undocumented:
 
-   Includes the content of module in the current
-   interactive module. Here module can be a module expression or a module
-   type expression. If module is a high-order module or module type
-   expression then the system tries to instantiate module by the current
-   interactive module.
+.. exn:: Signature components for label @ident do not match.
+    :undocumented:
 
-.. cmd:: Include {+<+ @module}
+.. exn:: The field @ident is missing in @qualid.
+   :undocumented:
 
-   is a shortcut for the commands :n:`Include @module` for each :token:`module`.
+.. |br| raw:: html
 
-.. cmd:: End @ident
+    <br>
 
-   This command closes the interactive module :token:`ident`. If the module type
-   was given the content of the module is matched against it and an error
-   is signaled if the matching fails. If the module is basic (is not a
-   functor) its components (constants, inductive types, submodules etc.)
-   are now available through the dot notation.
+.. note::
 
-    .. exn:: No such label @ident.
-       :undocumented:
+  #. Interactive modules and module types can be nested.
+  #. Interactive modules and module types can't be defined inside of :ref:`sections<section-mechanism>`.
+     Sections can be defined inside of interactive modules and module types.
+  #. Hints and notations (:cmd:`Hint` and :cmd:`Notation` commands) can also appear inside interactive
+     modules and module types. Note that with module definitions like:
 
-    .. exn:: Signature components for label @ident do not match.
-       :undocumented:
+     :n:`Module @ident__1 : @module_type := @ident__2.`
 
-    .. exn:: This is not the last opened module.
-       :undocumented:
+     or
 
-.. cmd:: Module @ident := @module_expression
+     :n:`Module @ident__1 : @module_type.` |br|
+     :n:`Include @ident__2.` |br|
+     :n:`End @ident__1.`
 
-    This command defines the module identifier :token:`ident` to be equal
-    to :token:`module_expression`.
+     hints and the like valid for :n:`@ident__1` are the ones defined in :n:`@module_type`
+     rather then those defined in :n:`@ident__2` (or the module body).
+  #. Within an interactive module type definition, the :cmd:`Parameter` command declares a
+     constant instead of definining a new axiom (which it does when not in a module type definition).
+  #. Assumptions such as :cmd:`Axiom` that include the :n:`Inline` clause will be automatically
+     expanded when the functor is applied, except when the function application is prefixed by ``!``.
```

Comment:
Looks good except that it can be confusing that the two successive notes do not refer to the same assumption command (`Parameter` / `Axiom`) even though these are equivalent. Not a big deal either.

Reply:
It's particularly odd to say "Parameter ... instead of defining a new **axiom**" if Parameters and Axioms are the same thing.  The equivalence deserves more explanation under the Parameter command.

The wording for Parameter should include Conjecture and Axiom.  And not clear if "Assumptions such as ..." include Variable and Hypothesis.

Please elaborate or I'll leave it as-is.

Reply:
Yes, these are the same thing, but yes, we tend to use a different terminology because they are not meant to be used in the same cases. But I think it is also fine to leave as is.